### PR TITLE
MQ and sda pipeline integration

### PR DESCRIPTION
**Pull request by blankdots**
### Description

This handles the Message Broker Integration and `sda-pipeline` in order to consume and publish necessary messages to finalize a submission.

### Related issues

part of #148
### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (this needs a follow up PR)

### Changes Made

- add MessageBroker class with MQPublisher  and MQConsumer separate classes functionality
- add cli tool for MQConsumer so that we can deploy multiple consumers independently of the web server - we can also go rpc as illustrated here: https://github.com/eandersson/amqpstorm/blob/2.x/examples/flask_threaded_rpc_client.py however i think as a design it is better to keep the consumer(s) separate from the web server
- add message broker publishing to workflow
- add rabbitmq + default config to integration tests
- update `.pre-commit-config.yaml`
- small fixes
- refactor `file.json` to keep track of versions
- add `version` `objectid` to file in submision

### Testing

- [x] Needs testing (start an issue or do a follow up PR about it)
- [x] Integration Tests (at least partly)

### Mentions

Files integration will be continued in a different PR, to make it easier to review
right now the integration with sda-pipeline is minimal however full integration will be handled in a different PR

### CODE REVIEW

File: metadata_backend/helpers/schemas/workflow.json [hunk 1]
```diff
@@ -137,7 +138,24 @@
...
+                },
+                "if": {
+                    "properties": {
+                        "name": {
+                            "const": "messageBroker"
+                        }
                     }
+                },
+                "then": {
+                    "required": [
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schemas/workflow.json:156]: not sure what you see it, however this is equivalent to saying if the `name` is `messageBroker` then add another required property to the ones listed on line 102 thus making the list `["name", "endpoint", "exchange"]`

**Comment by blankdots** [comment]
unit tests and full functionality for the files operator as well as sda-pipeline integration will need to be split in other PRs

**Review by csc-felipe — APPROVED** [review]
This is a big one. Mostly adding new code, which is not yet in use, so I expect we might need to adjust and write tests later.

On a side note, `scripts` folder starts to look funny, with lots of service configuration files. But I'm ok if we solve that some other time.

File: docs/dictionary/wordlist.txt [hunk 2]
```diff
@@ -650,9 +658,9 @@ subjectsschema
 submissionid
 submissionslice
 submissiontype
+submitter's
 submitterdemultiplexed
 submitterid
-submitter's
```
Comment (csc-jm) [inline docs/dictionary/wordlist.txt:655]: 🤨

Comment (blankdots) [reply docs/dictionary/wordlist.txt:655]: well yeah smth is off in the sorting, will revert that

Comment (csc-jm) [reply docs/dictionary/wordlist.txt:655]: If you keep it just like that, the sorting on my end will at least never change it again :D

Comment (blankdots) [reply docs/dictionary/wordlist.txt]: i think there is a difference between how vscode sorts and how the pre-commit hook `sort.sh` sorts

**Review by csc-jm — APPROVED** [review]

**Comment by blankdots** [comment]
merging this, as other things depend on it.

**Pull request merged**